diff --git a/node_modules/@remix-run/react/dist/esm/scroll-restoration.js b/node_modules/@remix-run/react/dist/esm/scroll-restoration.js
index 092cb17..34b07e2 100644
--- a/node_modules/@remix-run/react/dist/esm/scroll-restoration.js
+++ b/node_modules/@remix-run/react/dist/esm/scroll-restoration.js
@@ -125,6 +125,9 @@ function useScrollRestoration() {
         return;
       } // otherwise go to the top on new locations
 
+      if (location.state && typeof location.state === "object" && (location.state.scroll === false)) {
+        return;
+      }
 
       window.scrollTo(0, 0);
     }, [location]);
