diff --git a/node_modules/@remix-run/dev/dist/cli/commands.js b/node_modules/@remix-run/dev/dist/cli/commands.js
index 54f2d99..80eb2fa 100644
--- a/node_modules/@remix-run/dev/dist/cli/commands.js
+++ b/node_modules/@remix-run/dev/dist/cli/commands.js
@@ -228,6 +228,7 @@ async function watch(remixRootOrConfig, modeArg, callbacks) {
     },
 
     onRebuildFinish() {
+      purgeAppRequireCache(config$1.serverBuildPath);
       log(`Rebuilt in ${prettyMs__default["default"](Date.now() - start)}`);
       broadcast({
         type: "RELOAD"
@@ -289,7 +290,6 @@ async function dev(remixRoot, modeArg, portArg) {
   let app = express();
   app.disable("x-powered-by");
   app.use((_, __, next) => {
-    purgeAppRequireCache(config$1.serverBuildPath);
     next();
   });
   app.use(createApp(config$1.serverBuildPath, mode, config$1.publicPath, config$1.assetsBuildDirectory));
